src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -664,23 +664,25 @@ public ProjectState WithCompilationOptions(CompilationOptions options)
 
         var newProvider = new ProjectSyntaxTreeOptionsProvider(_analyzerConfigOptionsCache);
 
-        return With(projectInfo: ProjectInfo.WithCompilationOptions(options.WithSyntaxTreeOptionsProvider(newProvider))
+        return With(projectInfo: ProjectInfo.WithCompilationOptions(options?.WithSyntaxTreeOptionsProvider(newProvider))
                    .WithVersion(Version.GetNewerVersion()));
     }
 
-    public ProjectState WithParseOptions(ParseOptions options)
+    public ProjectState WithParseOptions(ParseOptions? options)
```

Comment:
curious what the case is where we set the parse options to null.

Reply:
You can call `WithProjectInfo` with null ParseOptions. Alternatively I can throw NotSupportedException. It was straightforward to support removing parse options.

Reply:
>  It was straightforward to support removing parse options.

tbh, it wasn't straightforward for me :)   i would far prefer us throwing until we have a use case where that is needed.

Reply:
ok. I'll throw

Reply:
thanks very much!